src/server.c

```diff
@@ -3209,6 +3209,10 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
             flushAppendOnlyFile(0);
     }
 
+    run_with_period(1000) {
+        if (server.aof_meta) delHistoryAofFilesCron(server.aof_meta);
+    }
```

Comment:
i haven't yet got to read the code behind it, but i wonder why this needs to be in cron, and not just when AOFRW succeeds or an AOF is being loaded after a crash?

Reply:
Yes, I actually thought about it. The `delHistoryAofFilesCron` function will use bio to delete all history files and update the manifest file on the disk simultaneously. In theory, it is not a very heavy operation, so let's move it to backgroundRewriteDoneHandler?

Reply:
There may be this possibility: the user exited the process before cleaning up the history (redis was killed), and the history files will be kept until the next rewrite success.

Reply:
this is why i suggested to call that on startup too.

Reply:
Modified, `delHistoryAofFiles` will be called in three places:
1. When the redis server starts
2. When AOFRW successfully finish
3. When `config set aof-enable-auto-gc yes`